docs/source/traitsui_user_manual/testing.rst

```diff
@@ -333,9 +333,27 @@ it with any testing framework (e.g. unittest, pytest).
 
 .. rubric:: Is UITester compatible with PyFace ModalDialogTester?
 
-Yes. For example, you can use |UITester| to launch a modal dialog, and use
+Yes, with some care.
+
+For example, you can use |UITester| to launch a modal dialog, and use
 ModelDialogTester to close it.
 
+But if you are attempting to modify or inspect GUI states using |UITester|
+while the dialog is opened, you should set the ``process_events`` attribute to
+false for those operations. Otherwise the ModalDialogTester and UITester
+will enter a race condition that blocks forever.
```

Comment:
The situation should be better described as a deadlock rather than a race condition.